devito/finite_differences/derivative.py

```diff
@@ -262,7 +262,10 @@ def __call__(self, x0=None, fd_order=None, side=None, method=None, **kwargs):
 
     def _rebuild(self, *args, **kwargs):
         kwargs['preprocessed'] = True
-        return super()._rebuild(*args, **kwargs)
+        if args:
+            return super()._rebuild(diffify(args[0]), *args[1:], **kwargs)
```

Comment:
This seems quite dangerous. Also `diffify` is usually not reconstructing things like `space_order` and such attributes properly so this will still loose a lot of information. 
You also probably better off doing that line 98

```
if not isinstance(expr, Differentiable):
    try:
        expr = diffify(expr)
    except not sure what:
        raise ValueError("`expr` must be a Differentiable object")
```

Reply:
I previously had that and I can't remember why I changed it, LMGBTY

Reply:
Moved, I potentially fixed the other issue elsewhere